descriptor.go

```diff
@@ -0,0 +1,25 @@
+package oras
```

Comment:
Can we put `descriptor.go` and `descriptor_test.go` in a `descriptor` package?

Reply:
Done.